#[
SyntaxError: 'except' type cannot have subtypes.
    from catch_generic.lly:13
]#

class Ex[A](message: string, v: A) < Exception(message)
{
    var @value = v
}

try: {
    raise Ex::new("", 100)
except Ex[integer] as e:
    if e.value != 100:
        stderr.print("Failed.")
}
